lib/DirectEditing/TextDirectEditor.php

```diff
@@ -83,7 +83,28 @@ public function getName(): string {
 	public function getMimetypes(): array {
 		return [
 			'text/markdown',
-			'text/plain'
+			'text/plain',
+			'application/cmd',
+			'application/x-empty',
+			'application/x-msdos-program',
+			'application/javascript',
+			'application/json',
+			'application/x-perl',
+			'application/x-php',
+			'application/x-tex',
+			'application/xml',
+			'application/yaml',
+			'text/css',
+			'text/csv',
+			'text/html',
+			'text/org',
+			'text/x-c',
+			'text/x-c++src',
+			'text/x-h',
+			'text/x-java-source',
+			'text/x-ldif',
+			'text/x-python',
+			'text/x-shellscript',
```

Comment:
Not sure if we should rather move those to be the optional opener for direct editing. I assume a local code editor would be more suitable for mobile editing if present.

@tobiasKaminsky What is the behavior on mobile if there is no local editor, would it fall back to the optional ones then or are those really only used through the context menu options?

Reply:
We do not differentiate between optional and normal ones, if there is no local app.
if there is a local app and file is downloaded, it is used.